Newdev

math-g opened the pull request:
Corrected broken pom.xml and made some updates on WOS full harvest example, mainly to make the Fetch work (root node is no longer ut but uid). Also updated the XSL template (concepts and publications datetimes were missing). For the xsl, I could also have added the publication day when available and the xsd datetimes datatype (^^xsd:datetime).

TheRedDev commented:
Ah, sorry, the comments on the commit probably should have been here ;)

fereira commented:
That’s a massive PR. It’s going to take me some time to review it (and I’m on a sprint this week).

From: grahamtriggs <notifications@github.com>
Reply-To: vivo-project/VIVO-Harvester <reply@reply.github.com>
Date: Monday, May 15, 2017 at 11:18 AM
To: vivo-project/VIVO-Harvester <VIVO-Harvester@noreply.github.com>
Cc: John Fereira <jaf30@cornell.edu>, Mention <mention@noreply.github.com>
Subject: Re: [vivo-project/VIVO-Harvester] Newdev (#36)


@grahamtriggs<https://github.com/grahamtriggs> requested your review on: vivo-project/VIVO-Harvester#36<https://github.com/vivo-project/VIVO-Harvester/pull/36> Newdev.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/vivo-project/VIVO-Harvester/pull/36#event-1082457356>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AAjs1MILAr_mbLOE2HkK3Bny2H6iKMY-ks5r6Gy0gaJpZM4NbEst>.

grahamtriggs commented:
That’s fine John - I’m not going to be able to look at it this week either :) Just wanted to ensure that you are aware of it.


On 15 May 2017 at 16:25:17, John Fereira (notifications@github.com) wrote:

That’s a massive PR. It’s going to take me some time to review it (and I’m on a sprint this week).

From: grahamtriggs <notifications@github.com>
Reply-To: vivo-project/VIVO-Harvester <reply@reply.github.com>
Date: Monday, May 15, 2017 at 11:18 AM
To: vivo-project/VIVO-Harvester <VIVO-Harvester@noreply.github.com>
Cc: John Fereira <jaf30@cornell.edu>, Mention <mention@noreply.github.com>
Subject: Re: [vivo-project/VIVO-Harvester] Newdev (#36)


@grahamtriggs<https://github.com/grahamtriggs> requested your review on: vivo-project/VIVO-Harvester#36<https://github.com/vivo-project/VIVO-Harvester/pull/36> Newdev.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/vivo-project/VIVO-Harvester/pull/36#event-1082457356>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AAjs1MILAr_mbLOE2HkK3Bny2H6iKMY-ks5r6Gy0gaJpZM4NbEst>.
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.

TheRedDev commented:
Ok, I think I understand why he wants the initAlgs to be public. The constructors take what it can produce, so someone using the harvester as a library could just create the map<string,string> without mucking around with converting them to algorithms. Fair enough.

TheRedDev commented:
Making Transfer.execute() public is no-harm-no-foul. I explain above why we thought it was unnecessary and had made it private, but I have no objection to making it public.

math-g commented:
Hi, sorry for the delay and the missing files (wos-query-message.xml + vivo.model.xml, bad manipulation from me trying to remove my specific settings). Actually, maybe it is not such a massive update. I just wanted to share what I had to modify to build the develop branch and make it harvest WOS data, especially updates in pom.xml and WOSFetch.java. See my comment about pom.xml.
Also publications datetimes in wos-to-vivo.datamap-v.1.6.xsl were missing and seemed to be needed to show the temporal graph. 
You are perfectly rignt about why I made the Score.java and Transfer.java constructors public. I needed it in my setup but you can ignore it.

math-g commented:
Hi @TheRedDev. It's done.

nateprewitt commented on example-scripts/bash-scripts/full-harvest-examples/1.0-1.5-examples/example-scopus/scopusfetch.config.xml:
```diff
@@ -9,7 +9,7 @@
 
 	<!-- Required: Scopus access info -->
-	<Param name="scopus-apikey">EnterAPIKeyHere</Param>
+	<Param name="scopus-apikey">cae8c3bac797762bf3de9c5c7f7c5c44</Param>
```
It seems like you may not have intended to commit an actual Scopus ID here.

nateprewitt commented on example-scripts/bash-scripts/full-harvest-examples/1.0-1.5-examples/example-scopus/vivo.model.xml:
```diff
@@ -102,8 +102,8 @@
     <Param name="modelName">http://vitro.mannlib.cornell.edu/default/vitro-kb-2</Param>
-    <Param name="dbUser">vitro</Param>
-    <Param name="dbPass">vitro123</Param>
+    <Param name="dbUser">vitrodbpsl</Param>
```
Was there a reason for changing these credentials?

TheRedDev commented on src/main/java/org/vivoweb/harvester/fetch/nih/ScopusFetch2.java:
```diff
@@ -0,0 +1,1483 @@
+	public static void main(String... args) {
+		Exception error = null;
+		try {
+			InitLog.initLogger(args, getParser("PubmedFetch", database));
```
PubmedFetch should be replaced with ScopusFetch2 (why 2?)

TheRedDev commented on src/main/java/org/vivoweb/harvester/fetch/nih/ScopusFetch2.java:
```diff
@@ -0,0 +1,1483 @@
+		Exception error = null;
+		try {
+			InitLog.initLogger(args, getParser("PubmedFetch", database));
+			log.info("PubmedFetch: Start");
```
Same

TheRedDev commented on src/main/java/org/vivoweb/harvester/fetch/nih/ScopusFetch2.java:
```diff
@@ -0,0 +1,1483 @@
+			log.debug("Stacktrace:",e);
+			error = e;
+		} finally {
+			log.info("PubmedFetch: End");
```
Same

math-g commented:
Thanks for your comments. I fixed it. I removed Scopus2 that is a work in progress specific to my institution. I replaced scopus config files with the original ones (api key should be IP specific). Sorry for the inconvenience.

TheRedDev commented:
@math-g not an inconvenience at all, thank you for contributing :)
Since the data is still there in the intermediate commits, we'll probably have to 'Squash' the commits so those files don't show up in the commit history.

TheRedDev commented:
@fereira I'll wait for you to look this over too. I haven't had a chance to actually run his code either, but last I checked WOSFetch was broken. I have a patch pending on WOSFetch as well that resolves some of the same things he fixed and has support for WOS Premium, but mine is mixed in with the huge Jena3 support overhaul I have going. If this fixes WOSFetch that is good since Jena3 patch is huge and not quite ready.

math-g commented:
@TheRedDev yes my fault, it's commited, Thanks if you can squash but otherwise don't worry that is not very sensitive.

TheRedDev commented:
@math-g if you know how, you can rebase and such to remove that from git history. Not certain how that will affect the pull request though.

fereira commented:
Thanks for all your help on the TheRedDev.   The PR looks fine overall now but I was wondering why the changes were made in bash-scripts/full-harvest-examples/1.0-1.5-examples.  That's the directory for harvesting data into a Vivo version 1.5 or earlier.   Has the wos-example there been tested on a VIVO v1.6 or later site.  If so, a copy of he directory should be put into the 1.6-examples directory.   I'll try to make some time this week to test out this PR.

TheRedDev commented:
I think that is because the 1.6+ examples folder does not have a wos example. Agreed that it should be moved to the 1.6+ folder, did not notice that.

math-g commented:
Hi. @TheRedDev I have rebased to skip the two files, PR seems to be ok (I hope so .. as it can perilous)
@fereira yes, as @TheRedDev said, in the develop branch I started on, the WOS script is adapted to 1.6 but is located in the 1.5 folder and I didn't change that.
